packages/vite/src/node/plugins/html.ts

```diff
@@ -323,6 +323,25 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
             config,
             publicToRelative,
           )
+        const nodeStartWithLeadingWhitespace = (
+          node: DefaultTreeAdapterMap['node'],
+        ) => {
+          if (node.sourceCodeLocation!.startOffset === 0)
+            return node.sourceCodeLocation!.startOffset
+
+          // Gets the offset for the start of the line including the
+          // newline trailing the previous node
+          const lineStartOffset =
+            node.sourceCodeLocation!.startOffset -
+            node.sourceCodeLocation!.startCol
+          const line = s.slice(
+            lineStartOffset,
+            node.sourceCodeLocation!.startOffset,
+          )
+          return /\S/.test(line)
```

Comment:
Maybe we can use `line.trim()` instead as it's more [performant](https://perf.link/#eyJpZCI6InQ5YXd3MjQ3a3FlIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IGRhdGEgPSBgbG9yZW0gaXBzdW0gZG9sb3Igc2l0IGFtZXQgY29uc2VjdGV0dXIgYWRpcGlzaWNpbmcgZWxpdC4gUXVpc3F1YW0sIHF1b3NgLnNwbGl0KCcnKS5tYXAocyA9PiBzLnJlcGVhdCg1KSkiLCJ0ZXN0cyI6W3sibmFtZSI6InRyaW0iLCJjb2RlIjoiZm9yIChjb25zdCBkIG9mIGRhdGEpIHtcbiAgZC50cmltKClcbn0iLCJydW5zIjpbMTYyMDAwLDE2MjAwMCwyNzgwMDAsMjQxMDAwLDI2MTAwMCwxODIwMDAsMzIwMDAwLDE0MjAwMCw1MjAwMCwyMTMwMDAsOTgwMDAsMTk2MDAwLDE5NTAwMCw2MTAwMCwxMjMwMDAsOTAwMCw2NzAwMCwxOTMwMDAsOTIwMDAsMTg4MDAwLDE3MTAwMCwxMDkwMDAsMTQ4MDAwLDE3MjAwMCwxODcwMDAsMjE3MDAwLDgwMDAsMjQwMDAwLDEzMDAwMCw2MDAwMCw1NDMwMDAsMTM3MDAwLDE5ODAwMCwyMDgwMDAsMTM2MDAwLDE4MDAwLDM0MDAwMCw3OTAwMCw3NTAwMCwxNTQwMDAsMjYwMDAsNzQwMDAsMjEyMDAwLDEyNDAwMCw2ODAwMCw3ODAwMCw2MDQwMDAsMTk4MDAwLDE1NzAwMCwxMDIwMDAsOTQwMDAsMjk5MDAwLDkwMDAsMTczMDAwLDk1MDAwLDU1MDAwLDE1NzAwMCwxNDcwMDAsMTExMDAwLDE0NjAwMCwxMTAwMCwyMjkwMDAsMTU5MDAwLDUxMjAwMCwyNjgwMDAsMTUzMDAwLDMwODAwMCw3MzAwMCwxNjYwMDAsMTYwMDAwLDE2OTAwMCwzNTEwMDAsODAwMCw5MDAwLDIwMjAwMCwyNDAwMCwyMDMwMDAsODkwMDAsOTcwMDAsMTE2MDAwLDE3OTAwMCw5MDAwLDEyMzAwMCwyODcwMDAsMTE3MDAwLDkzMDAwLDQzMjAwMCw5MjAwMCwxMzgwMDAsMjQ4MDAwLDk0MDAwLDEwNDAwMCwxNTMwMDAsMjEzMDAwLDE3OTAwMCwyMTUwMDAsMjM4MDAwLDE0NTAwMCwxODcwMDAsMTYwMDAwXSwib3BzIjoxNjIwNzB9LHsibmFtZSI6InJlZ2V4IiwiY29kZSI6ImZvciAoY29uc3QgZCBvZiBkYXRhKSB7XG4gIC9cXFMvLnRlc3QoZClcbn0iLCJydW5zIjpbMzMwMDAsMjgwMDAsMTAwMDAsMzYwMDAsOTkwMDAsNTMwMDAsMTAwMCw1MjAwMCwxODAwMCwzMTAwMCwyNTUwMDAsMTYwMDAsOTMwMDAsNTYwMDAsNDEwMDAsMjYyMDAwLDIwMDAwLDQ5MDAwLDE0MDAwLDI5MDAwLDMwMDAsMTEwMDAsMjIwMDAsNDUwMDAsMzAwMDAsNjAwMCwxMzQwMDAsMzgwMDAsMjAwMDAsNTAwMDAsMzAwMCwyOTAwMCw3MjAwMCwyMzAwMCwzNjAwMCwxMDAwLDE2MzAwMCwzNzAwMCwyODAwMCwzNjAwMCw4MDAwMCwzMDUwMDAsMzUwMDAsMjkwMDAsNjEwMDAsMTAwMCwzMDAwLDQ5MDAwLDM2MDAwLDk0MDAwLDMwMDAwLDE4NzAwMCw4MDAwLDY4MDAwLDEwMDAwMCwyMjAwMCwyODAwMCw0NTAwMCw0MTAwMCwxNDAwMCwyMDAwLDk5MDAwLDMxMDAwLDEwMDAsMzYwMDAsMTAwMDAsMTU4MDAwLDEzMDAwLDcyMDAwLDMzMDAwLDEzMDAwLDE3MzAwMCwxMDAwLDgwMDAsNjYwMDAsMTEwMDAsNDkwMDAsNzUwMDAsMTEwMDAsMjcwMDAsMjEwMDAsMjUxMDAwLDI2MDAwLDEyMDAwLDI3MDAwLDIzMDAwLDIyMTAwMCwxMDAwLDI5MDAwLDkxMDAwLDM3MDAwLDM2MDAwLDgwMDAsNzkwMDAsNTUwMDAsMTkwMDAsMTM5MDAwLDE1MDAwLDEzMDAwLDcyMDAwXSwib3BzIjo1MjgzMH0seyJuYW1lIjoicmVnZXggKGNhY2hlZCkiLCJjb2RlIjoibGV0IHJlID0gL1xcUy9cbmZvciAoY29uc3QgZCBvZiBkYXRhKSB7XG4gIHJlLnRlc3QoZClcbn0iLCJydW5zIjpbNTgwMDAsMzEwMDAsMTAwMCw4NzAwMCwxNjMwMDAsMTI2MDAwLDI1NzAwMCw3NjAwMCwzMjAwMCw3NjAwMCwxMTAwMCw0MDUwMDAsOTMwMDAsMjAwMDAsNDIwMDAsMTAwMCwxNjAwMCwzNTAwMCwxMTAwMCwxOTAwMCw3MDAwLDEwMDAwLDkwMDAsODEwMDAsMjQwMDAsMjUwMDAsMTkxMDAwLDkzMDAwLDI2MDAwLDIzMDAwLDEwMDAsMTA0MDAwLDk1MDAwLDE0MDAwLDUyMDAwLDgwMDAsMjAyMDAwLDMyMDAwLDIyMDAwLDM4MDAwLDEyMDAwLDQxMzAwMCw2OTAwMCw1MDAwMCwxMDAwMCw3MDAwLDI2MTAwMCwxMzAwMCwzNjAwMCw4NTAwMCwyMDAwLDIxMTAwMCw0MDAwLDYwMDAwLDEyMDAwLDQ2MDAwLDQyMDAwLDYzMDAwLDQ1MDAwLDcwMDAsMTAwMCwxMjkwMDAsMTEwMDAsMzY5MDAwLDQ2MDAwLDExMDAwLDE2OTAwMCwxODAwMCw5NzAwMCwzNzAwMCw4MDAwLDIxOTAwMCwxMDAwMCwxMDAwLDMxMDAwLDM3MDAwLDEwNTAwMCwxNzAwMCwxOTAwMCwxMDMwMDAsMTEwMDAsMjY0MDAwLDI2MDAwLDcwMDAsODAwMDAsNjIwMDAsMjkyMDAwLDgwMDAsMzQwMDAsNTIwMDAsNTgwMDAsMzkwMDAsMTkwMDAsODAwMDAsMTAwMDAsMTcwMDAsMjMwMDAwLDI1MDAwLDIxMDAwLDY5MDAwXSwib3BzIjo2OTA3MH1dLCJ1cGRhdGVkIjoiMjAyMy0xMC0wNVQwOTowMjoyOS44NDJaIn0%3D)?

Reply:
I suppose? Personally, I'd find that to be quite unclear and the performance gain is non-existent for this specific use case. You'd need thousands of tags to make a measurable difference.

If that's what you want though I'll switch it over.

Reply:
It also took me a while to understand what `/\S/` does 😅 I'd prefer `.trim()` and also put a comment above to explain what we're doing. Every little performance counts and they stack over time if we're not careful.

Reply:
I don't love `\S` either tbh, but that's on your linter! 😅 I believe it forced `[^\s]` to that, which, fair enough, though I wasn't familiar with `\S` myself.

Will update sometime tomorrow, pretty late here for me.

Reply:
Sorry, took a bit longer to get to this. Switched over the detection and added comments hopefully clarifying which situations are being covered, lmk if you'd like to see any further revisions.